src/NuGetGallery/Controllers/PackagesController.cs

```
@@ -434,6 +434,22 @@ public virtual JsonResult UploadPackageProgress()
 
             var model = new DisplayPackageViewModel(package, packageHistory);
 
+            if (package.HasReadMe ?? false)
+            {
+                try
+                {
+                    //Reads the README file and push to the view
+                    using (var reader = new StreamReader(await _packageFileService.DownloadReadmeFileAsync(package, Constants.HtmlFileExtension), Encoding.UTF8))
+                    {
+                        model.ReadMeHtml = await reader.ReadToEndAsync();
+                    }
+                }
+                catch (Exception)
+                {
+                    model.ReadMeHtml = null;
+                }
+            }
+
             if (package.IsOwner(User))
```

In case the readme load fails you should show an error page (i.e. let the exception propagate). We shouldn't hide the error.